scripts/job_manager.ps1

```diff
@@ -0,0 +1,94 @@
+class MyJobManager {
+    [String] $Name = "JobManager"
+    [Hashtable] $Jobs = @{}
+
+    JobManager () {
+    }
+
+    [void] AddJob ($Tracker, $ScriptBlock, $ArgumentList, $InitScript) {
+        $job = Start-Job -ScriptBlock $ScriptBlock `
+                         -ArgumentList $ArgumentList `
+                         -InitializationScript $InitScript
+        Start-Sleep -Milliseconds 100
+        if ($job.State -ne "Running" ) {
+            Write-Host "Job failed to start" -ForegroundColor Red
+            $output = Receive-Job -Job $job
+            Remove-Job -Job $job
+            Write-host "Job failed with output `r`n$output`r`n" -ForegroundColor Red
+            return
+        }
+        if ($this.Jobs.Contains($Tracker)) {
+            $this.Jobs[$Tracker].AddLast($job)
+        } else {
+            $this.Jobs[$Tracker] = New-Object Collections.Generic.LinkedList[object]
+            $this.Jobs[$Tracker].AddLast($job)
+        }
+    }
+
+    [void] RemoveTracker ($Tracker) {
+        Write-Host "Removing jobs from tracker: $Tracker"
+        Start-Sleep 5
+        $current = $this.Jobs[$Tracker].First
+        while(-not ($current -eq $null)) {
+            Remove-Job $current.Value
+            $current = $current.Next
+        }
+        $this.Jobs.Remove($Tracker)
+    }
+
+    [Collections.Generic.LinkedList[object]] GetJobsFromTracker ($Tracker) {
+        if ($this.Jobs.Contains($Tracker)) {
+            return $this.Jobs[$Tracker]
+        } else {
+            Write-Host "no tracker"
+            return $null
+        }
+    }
+
+    [String] WaitForJobCompletion ($Tracker, $Timeout ) {
+        Write-Host "Waiting for Jobs to finish in Tracker : $Tracker"
+        $status = "running"
```

Comment:
look bellow

Reply:
when waiting for job, it only makes sense to wait for them if they are in the running state
no need to change this method